bindings/go/oci/credentials/docker_config.go

```diff
@@ -0,0 +1,179 @@
+	}
+	credentialMap := map[string]string{}
+	if v := cred.Username; v != "" {
+		credentialMap[CredentialKeyUsername] = v
+	}
+	if v := cred.Password; v != "" {
+		credentialMap[CredentialKeyPassword] = v
+	}
+	if v := cred.AccessToken; v != "" {
+		credentialMap[CredentialKeyAccessToken] = v
+	}
+	if v := cred.RefreshToken; v != "" {
+		credentialMap[CredentialKeyRefreshToken] = v
+	}
+
+	return credentialMap, nil
+}
+
+// getStore creates a credential store based on the provided Docker configuration.
+// It supports three modes of operation:
+//   - Default mode: Uses system default Docker config locations
+//   - Inline config: Uses a provided JSON configuration string
+//   - File-based: Uses a specified Docker config file
+//
+// The function handles:
+//   - Shell expansion for file paths (e.g., ~ for home directory)
+//   - Temporary file creation for inline configurations
+//   - Integration with the native host credential store
+func getStore(ctx context.Context, dockerConfig credentialsv1.DockerConfig) (remotecredentials.Store, error) {
+	var store remotecredentials.Store
+	var err error
+	switch {
+	case dockerConfig.DockerConfigFile == "" && dockerConfig.DockerConfig == "":
+		slog.InfoContext(ctx, "attempting to load docker config from default locations or native host store")
+		store, err = remotecredentials.NewStoreFromDocker(remotecredentials.StoreOptions{
+			DetectDefaultNativeStore: true,
+		})
+	case dockerConfig.DockerConfig != "":
+		slog.InfoContext(ctx, "using docker config from inline config")
+		var tmp *os.File
+		if tmp, err = os.CreateTemp("", "docker-config-*.json"); err != nil {
+			return nil, fmt.Errorf("failed to create temporary file: %w", err)
+		}
+		defer func() {
+			_ = tmp.Close()
+			_ = os.Remove(tmp.Name()) // we can safely remove the temp file because the store is loaded into memory
+		}()
+		if err := os.WriteFile(tmp.Name(), []byte(dockerConfig.DockerConfig), 0o600); err != nil {
+			return nil, fmt.Errorf("failed to write temporary file: %w", err)
+		}
+		store, err = remotecredentials.NewStore(tmp.Name(), remotecredentials.StoreOptions{})
+	case dockerConfig.DockerConfigFile != "":
+		slog.InfoContext(ctx, "using docker config from file", "file", dockerConfig.DockerConfigFile)
+		path := dockerConfig.DockerConfigFile
+		// old OCM versions did unclean shell expansion...
+		// simple shell expansion for legacy compatibility, does not support cross user expansion
+		if idx := strings.Index(path, "~"); idx != -1 {
+			dirname, err := os.UserHomeDir()
+			if err != nil {
+				return nil, fmt.Errorf("failed to get user home directory: %w", err)
+			}
+			path = strings.ReplaceAll(path, "~", dirname)
+		}
+		if _, err := os.Stat(path); err != nil {
+			slog.WarnContext(ctx, "failed to find docker config file, thus the config will not offer any credentials", "path", path)
+		}
+		store, err = remotecredentials.NewStore(path, remotecredentials.StoreOptions{})
```

Comment:
Maybe add a small comment ( so others don't have to do what I did which is check if NewStore throws an error on none existing config ) that `NewStore` returns no error on `if os.IsNotExist(err) {`.

Reply:
good point, added comment accordingly